libvcs4j-spoon/src/main/java/de/unibremen/informatik/st/libvcs4j/spoon/SpoonModel.java

```diff
@@ -0,0 +1,336 @@
+
+            filesNotCompiledSinceLastUpdate.addAll(getAllFilesFromFileChanges(revisionRange.getAddedFiles(), false));
+
+            launcher.addInputResource(createInputSource(filesNotCompiledSinceLastUpdate));
+            launcher.getModelBuilder().addCompilationUnitFilter(path -> !filesToBuild.contains(path));
+            launcher.getModelBuilder().compile(SpoonModelBuilder.InputType.FILES);
+            model.get().setBuildModelIsFinished(false);
+            try {
+                model = Optional.of(launcher.buildModel());
+            } catch (Exception e) {
+                model = Optional.empty();
+                filesNotCompiledSinceLastUpdate.clear();
+            }
+
+        } else {
+
+            final Launcher launcher = new Launcher();
+            launcher.setBinaryOutputDirectory(temporaryDirectory);
+            launcher.getEnvironment().setNoClasspath(noClasspath);
+            //Add the checked out directory here, so we do not have to add each single file
+            launcher.addInputResource(revisionRange.getRevision().getOutput().toString());
+            launcher.getModelBuilder().compile(SpoonModelBuilder.InputType.FILES);
+            model = Optional.of(launcher.buildModel());
+
+        }
+        LOGGER.info(format("Model built in %d milliseconds", currentTimeMillis() - current));
+    }
+
+    /**
+     * Transforms the given Input into a {@link FilteringFolder}. Each ath in the given collection has to exist,
+     * be readable and be a regular Java source file, otherwise it will not be added to the {@link FilteringFolder}.
+     *
+     * @param input The given collection with absolute paths to source files.
+     * @return A {@link FilteringFolder} containing all files from the given input.
+     */
+    private FilteringFolder createInputSource(final Collection<String> input) {
+        FilteringFolder folder = new FilteringFolder();
+        input.forEach(path -> {
+            final Path p = Paths.get(path);
+            if (Files.exists(p) && Files.isReadable(p) && Files.isRegularFile(p)) {
+                folder.addFile(new FileSystemFile(p.toFile()));
+            }
+        });
+        return folder;
+    }
+
+    /**
+     * Removes all source files from {@link CtModel} that are given as input. These files have changed, so they need to be
+     * removed from the {@link CtModel}. The corresponding binary files are getting deleted as well.
+     *
+     * @param pFileChanges The given collection with absolute paths to source files.
+     */
+    private void removeChangedTypes(final Collection<String> pFileChanges) {
+        final CtPackage rootPackage = model.get().getRootPackage();
+        final CompilationUnitFactory cuFactory = rootPackage.getFactory().CompilationUnit();
+
+        pFileChanges.stream()
+                .filter(sourceFile -> sourceFile.endsWith(".java"))
+                .forEach(path -> {
+                    if (cuFactory.getMap().containsKey(path)) {
```

Comment:
You should also delete packages which became empty after deleting their sub packages and files.